awx/api/views/__init__.py

```diff
@@ -4457,7 +4457,8 @@ class WorkflowApprovalApprove(RetrieveAPIView):
 
     def post(self, request, *args, **kwargs):
         obj = self.get_object()
-        request.user.can_access(models.WorkflowApproval, 'approve_or_deny', obj)
+        if not request.user.can_access(models.WorkflowApproval, 'approve_or_deny', obj):
+            raise PermissionDenied(detail=_("User does not have permission to approve or deny this workflow."))
```

Comment:
@beeankha oooooookay, I guess this explains why we saw the HTTP 403 the other day - it was the `ModelPermission` implementation.  _This_ makes more sense to me now, and should actually return the custom message in the HTTP response 👍